Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -475,6 +476,21 @@ DirectoryBlobReader::DirectoryBlobReader(
   }
 }
 
+DirectoryBlobReader::DirectoryBlobReader(const DirectoryBlobReader& rhs)
+    : m_gamecube_pseudopartition(rhs.m_gamecube_pseudopartition),
+      m_nonpartition_contents(rhs.m_nonpartition_contents), m_encryption_cache(this),
+      m_is_wii(rhs.m_is_wii), m_encrypted(rhs.m_encrypted),
+      m_disc_header_nonpartition(rhs.m_disc_header_nonpartition),
+      m_partition_table(rhs.m_partition_table), m_wii_region_data(rhs.m_wii_region_data),
+      m_extra_data(rhs.m_extra_data), m_data_size(rhs.m_data_size),
+      m_wrapped_volume(CreateDisc(rhs.m_wrapped_volume->GetBlobReader().CopyReader()))
```

Comment:
Now that you're using shared pointers, `m_disc_header_nonpartition`, `m_wii_region_data`, and `m_partition_table` can be turned from class members into local variables, and `m_extra_data` can be removed entirely. The only purpose of these member variables was to ensure that the data wasn't deallocated.